### PR TITLE
JAMES-2404 Register Quota indexer as a configured listener

### PR DESCRIPTION
**Pull request by chibenwa**
This allow more flexibily to the administrator. For example registering
this as an asynchronous component...

### CODE REVIEW

File: server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/MailboxListenersLoaderImpl.java [hunk 1]
```diff
@@ -88,6 +93,15 @@ public MailboxListener createListener(ListenerConfiguration configuration) {
         }
     }
 
+    public Optional<HierarchicalConfiguration> getConfiguration(ListenerConfiguration configuration) {
+        return OptionalUtils.or(
+            configurationOverrides.stream()
```
Comment (mbaechler) [inline server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/MailboxListenersLoaderImpl.java]: please extract this part

File: server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/MailboxListenersLoaderImpl.java [hunk 2]
```diff
@@ -88,6 +93,15 @@ public MailboxListener createListener(ListenerConfiguration configuration) {
...
+                .filter(override -> override.getClazz().getName().equals(configuration.getClazz()))
```
Comment (mbaechler) [inline server/container/guice/mailbox/src/main/java/org/apache/james/modules/mailbox/MailboxListenersLoaderImpl.java]: this test is weird : you compare getName() return to getClazz() return

File: src/site/xdoc/server/config-listeners.xml [hunk 3]
```diff
@@ -61,7 +61,9 @@
                  <li><code>org.apache.james.mailbox.quota.mailing.listeners.QuotaThresholdCrossingListener</code>:
                      For Cassandra guice wiring. Sends emails to users exceeding 80% and 99% of their quota to warn them</li>
                  <li><code>org.apache.james.quota.search.elasticsearch.events.ElasticSearchQuotaMailboxListener</code>:
-                     For Cassandra guice wiring. Index quota to allow to search user in a given ratio occupation range</li>
+                     For Cassandra guice wiring. Index quota to allow to search user in a given ratio occupation range.<br/>
+                     This mailet is asynchronous by default, but this behaviour can be overridden by the <b>async</b>
```
Comment (mbaechler) [inline src/site/xdoc/server/config-listeners.xml]: it looks synchronous to me

File: mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinConfiguration.java [hunk 4]
```diff
@@ -22,11 +22,35 @@
...
+    public static SpamAssassinConfiguration fromXML(HierarchicalConfiguration configuration) {
+        Optional<Host> host = Optional.ofNullable(configuration.getString(HOST_KEY, null))
+            .map(Host::parseConfString);
+
+        if (!host.isPresent()) {
+            LOGGER.warn("Could not find 'host' property. Disabling this service.");
+        }
+
+        return new SpamAssassinConfiguration(
+            host);
```
Comment (mbaechler) [inline mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinConfiguration.java]: should be on the previous line

File: mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinConfiguration.java [hunk 5]
```diff
@@ -22,11 +22,35 @@
...
+    }
+
+    public static HierarchicalConfiguration generateXMLForHost(Host host) {
```
Comment (mbaechler) [inline mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassinConfiguration.java]: do we really want to generate XML ?

**Comment by chibenwa** [comment]
I understand your point.

Quota indexing is compulsory if we expose the corresponding configured routes.

Hence I will close this PR.

**Pull request closed without merging**